File: algo/src/main/java/com/hartwig/actin/algo/doid/datamodel/Metadata.java
package com.hartwig.actin.algo.doid.datamodel;

import java.util.List;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(allParameters = true,
             passAnnotations = { NotNull.class, Nullable.class })
public abstract class Metadata {

    @Nullable
    public abstract Definition definition();

    @Nullable
    public abstract List<String> subsets();

    @Nullable
    public abstract List<Xref> xrefs();

    @Nullable
    public abstract List<Synonym> synonyms();

    @Nullable
    public abstract List<BasicPropertyValue> basicPropertyValues();

    @Nullable
    public abstract String snomedConceptId();

}

File: common/src/main/java/com/hartwig/actin/doid/datamodel/Metadata.kt
package com.hartwig.actin.doid.datamodel

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class Metadata {
    abstract fun definition(): Definition?
    abstract fun subsets(): List<String?>?
    abstract fun xrefs(): List<Xref?>?
    abstract fun synonyms(): List<Synonym?>?
    abstract fun basicPropertyValues(): List<BasicPropertyValue?>?
    abstract fun snomedConceptId(): String?
    abstract fun deprecated(): Boolean?
    abstract fun comments(): List<String?>?
}
